app/src/main/java/com/anysoftkeyboard/dictionaries/sqlite/AutoDictionary.java

```diff
@@ -69,7 +69,7 @@ public boolean addWord(String word, int frequencyDelta) {
             int freq = getWordFrequency(word);
 
             freq = freq < 0 ? frequencyDelta : freq + frequencyDelta;
-            if (freq >= mLearnWordThreshold) {
+            if (mLearnWordThreshold != -1 && freq >= mLearnWordThreshold) {
```

Comment:
Do we have any test that verifies this?

Reply:
I don't know. I tested it manually... I'm not familiar with automated tests.

I just sent a commit to fix a test because you wrote me how to.